substratest/client.py

```diff
@@ -85,9 +85,34 @@ def add_testtuple(self, spec):
 
     def add_compute_plan(self, spec):
         res = self._client.add_compute_plan(spec.to_dict())
+        # The API outputs of the create and get/list methods differ at the
+        # moment. This harmonizes the returns so that they can be parsed.
+        # See:
+        # https://github.com/SubstraFoundation/substra-chaincode/issues/36
+        res['traintuples'] = res['traintupleKeys']
+        res['testtuples'] = res['testtupleKeys']
+        del res['traintupleKeys']
+        del res['testtupleKeys']
+        res['algoKey'] = spec.algo_key
+        res['objectiveKey'] = spec.objective_key
+
         cp = assets.ComputePlan.load(res)
         return cp
 
+    def list_compute_plan(self, *args, **kwargs):
+        res = self._client.list_compute_plan(*args, **kwargs)
+        return [assets.ComputePlan.load(x) for x in res]
+
+    def get_compute_plan(self, *args, **kwargs):
+        res = self._client.get_compute_plan(*args, **kwargs)
+        compute_plan = assets.ComputePlan.load(res)
+        # the testtuples key of the compute_plan may be null instead
+        # of [] but should really be []
+        # See:
+        # https://github.com/SubstraFoundation/substra-chaincode/issues/36
+        compute_plan.testtuples = compute_plan.testtuples or []
```

Comment:
This should be done inside the `assets.py` module. We could maybe overwrite the `__init__` method or using `default_factory=list`?

Reply:
Thanks for pointing out the `default_factory=list`, it gave me the occasion of looking into it. It doesn't solve the issue here because it only provides a default if no value is provided at all.

Here is an example:

```python
>>> @dataclass
... class C:
...     mylist: List[int] = field(default_factory=list)
...

>>> c = C()
>>> c.mylist == []
True
>>> c = C(mylist=None)
>>> c.mylist is None
True
```

Reply:
I found another solution btw, it's coming up soon :)